### ISSUE TITLE
suggestion: add more options for naming conventions to the summary_frame methods

### ISSUE THREAD

**Issue by cpbl** [statsmodels/statsmodels#3314]
As follow-up to #3309, 
could we add some options for naming conventions to the summary_frame methods.
For instance, in the case of `MNLogit`, the `endog` column looks somethign like what's below:

```
> me.summary_frame().reset_index().head()
   endog        exog     dy/dx  Std. Err.         z      Pr(>|z|)  \
0  myvar=0      age100  0.134127   0.017388  7.713880  1.220494e-14   
1  myvar=0    age100sq -0.128643   0.016293 -7.895520  2.891064e-15   
2  myvar=0     married -0.002546   0.000580 -4.389987  1.133577e-05   
3  myvar=0        male  0.000640   0.000488  1.312193  1.894551e-01   
4  myvar=0  lnHHincome -0.003306   0.000346 -9.565278  1.119005e-21   
```

It might in this case be convenient for many users to have instead a column labeled `myvar` simply taking the values that appear after the "=" sign in the existing "endog" column.

There are, points out @josef-pkt in #3309, other multivariate endog cases which need  automatic name creation, e.g. "when we need the raveled params and associated statistics."

I doubt following is helpful, since the key parts would get done in other helper methods, but I wrapped summary_frame() as follows to get my column named "myvar":
```
def _mlogit_marginal_effects_summary_frame(me): 
    df=me.summary_frame().reset_index()
    endog=me.results._get_endog_name(None,None)[0]
    df[endog] = pd.Series(  df.endog.str.split('=').apply(pd.Series)[1])
    return(df)
```

**Comment by josef-pkt** [comment]
based on briefly browsing the code 
`MultinomialResults._get_endog_name`  creates the `'='.join([yname, name])` combination.
I think adding a keyword to return just the level names instead might be enough.

**Comment by cpbl** [comment]
Yes, changing the column name is easy enough for users to do.. Thanks.
